src/AutoRest.CSharp/Common/Input/InputTypes/InputDateTimeType.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace AutoRest.CSharp.Common.Input;
+
+internal record InputDateTimeType(DateTimeKnownEncoding Encode, InputPrimitiveType WireType, bool IsNullable) : InputType("DateTime", IsNullable) // TODO -- name?
+{
+    public static InputDateTimeType Rfc3339 { get; } = new(DateTimeKnownEncoding.Rfc3339, InputPrimitiveType.String, false);
+    public static InputDateTimeType Rfc7231 { get; } = new(DateTimeKnownEncoding.Rfc7231, InputPrimitiveType.String, false);
+    public static InputDateTimeType UnixTimestamp { get; } = new(DateTimeKnownEncoding.UnixTimestamp, InputPrimitiveType.Int64, false);
```

Comment:
UnixTimeStamp can use `int32` or `float` as `WireType`.  So it is not good enough to define a static instance `UnixTimestamp` with `Int64`

Reply:
This is no case that this is float - unix time stamp should always be integers
How about I rename this to `IntUnixTimestamp` and a new `LongUnixTimestamp`?

Reply:
And this is only a shortcut for us when we want to instantiate an instance of this. I think if we could make the name more descriptive to avoid a mistake when we use them, it should be fine.
The majority of the instance of this class should come from deserialization. We usually only use these static instances in the `CodeModelConverter` where we convert swaggers into input types.

Reply:
I changed my mind a little bit because we only handle the case with unix time stamp with long, therefore I will not add a static property here for IntUnixTimestamp

Reply:
these helpers are removed.